pytential/symbolic/primitives.py

```diff
@@ -347,9 +372,26 @@
     )
 
 
-Operand = Union["Expression", np.ndarray, MultiVector]
+# {{{ helpers
+
+
+@expr_dataclass()
+class Expression(ExpressionBase):
+    """A subclass of :class:`pymbolic.primitives.Expression` for use with
+    :mod:`pytential` mappers.
+    """
+
+    def make_stringifier(self, originating_stringifier=None):
+        from pytential.symbolic.mappers import StringifyMapper
+        return StringifyMapper()
+
+
+Operand: TypeAlias = (
+    ArithmeticExpressionT | np.ndarray[Any, np.dtype[Any]] | MultiVector)
 QBXForcedLimit = int | Literal["avg"] | None
 
+ExpressionT = TypeVar("ExpressionT", bound=ArithmeticExpressionT)
```

Comment:
This needs a bit of thought? I'm not sure our naming scheme is sufficiently smart :grin:

Reply:
I'm not sure I understand the issue. Could you explain?

Reply:
I just meant that it's a typevar bound to `ArithmeticExpressionT` and it's called `ExpressionT` + it's different to the `ExpressionT` in pymbolic.

Reply:
OIC. Good question... Addressing this properly would amount to establishing a new convention, which I'm kind of hesitant about. I guess we sort of had a convention of naming type variables `SomethingT`, but then I named `ExpressionT` the way I did... because `Expression` was taken. :thinking: PEP8 has [guidance](https://pep8.org/#type-variable-names) on naming type variables, but nothing that would help in this case. Do you have an opinion?

Reply:
Yeah, I've also mostly gone with naming type variables as `SomethingT`, but besides that, can't say I have any useful ideas. 

Maybe `ExpressionLike` would have worked instead of `ExpressionT`? Since it could also potentially fit `MultiVector` and other things that mainly mappers know how to handle. `ExpressionNode` to emphasize the dag part?

Reply:
Is renaming `ExpressionT` and/or `ArithmeticExpressionT` on the table? It's definitely way easier to just rename this one variable and not break the world :))

Reply:
We'd have to probably leave it in place deprecated for a bit, because the type names are referred to in `cast`s. I'm not sure I love `ExpressionLike`. Maybe `ExpressionOrData` for `ExpressionT`?  And just `ArithmeticExpression` for `ArithmeticExpressionT`?

Reply:
> I'm not sure I love ExpressionLike. Maybe ExpressionOrData for ExpressionT?

Can't say I have a big preference there.. I'll try to keep it in the back of my head and see if anything comes to mind :\

> And just ArithmeticExpression for ArithmeticExpressionT?

Yep!

In the meantime, I've renamed the one here to not be completely confusing.

Reply:
For posterity: this discussion became https://github.com/inducer/pymbolic/pull/162.